yt/frontends/ramses/data_structures.py

```diff
@@ -197,13 +214,85 @@ def fill(self, fd, fields, selector, file_handler):
         # Initializing data container
         for field in fields:
             tr[field] = np.zeros(cell_count, 'float64')
-
         fill_hydro(fd, file_handler.offset,
-                   file_handler.level_count, levels, cell_inds,
+                   file_handler.level_count,
+                   [self.domain_id-1],
+                   levels, cell_inds,
                    file_inds, ndim, all_fields, fields, tr,
                    oct_handler)
         return tr
 
+    def _fill_with_ghostzones(self, fd, fields, selector, file_handler, num_ghost_zones):
+        ndim = self.ds.dimensionality
+        ncpu = self.ds.parameters['ncpu']
+        # Here we get a copy of the file, which we skip through and read the
+        # bits we want.
+        oct_handler = self.oct_handler
+        all_fields = [f for ft, f in file_handler.field_list]
+        fields = [f for ft, f in fields]
+        tr = {}
```

Comment:
In fairness to @cphyc we do use `tr` as "to return" in lots of places.  I take the blame for that.

Reply:
So I'm torn between changing `tr`  here and break the "convention", or keep it... I'm leaving as is for now, unless someone argues to change it (in which case we should probably grep `tr` and replace it at other places as well).

Reply:
I'd be in favour of changing it since I managed to avoid this convention for more than a year of reading/writing for yt, so it really doesn't seem obvious to me, and I'm all about reducing overhead on the dev side :)
In any case I don't want to cause any more hurdle than necessary for this PR, so imo this can stay as is.

Reply:
Done!